src/components/notifications/notifications_dropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { HTMLAttributes } from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+import { setHasUnreadNotifications } from '../../store/actions';
+import { getHasUnreadNotifications, getNotifications } from '../../store/selectors';
+import { themeColors, themeDimensions } from '../../util/theme';
+import { Notification, StoreState } from '../../util/types';
+import { CardBase } from '../common/card_base';
+import { Dropdown, DropdownPositions } from '../common/dropdown';
+import { BellIcon } from '../common/icons/bell_icon';
+
+import { NotificationItem } from './notification_item';
+
+interface StateProps {
+    notifications: Notification[];
+    hasUnreadNotifications: boolean;
+}
+interface DispatchProps {
+    onMarkNotificationsAsRead: () => any;
+}
+
+type Props = HTMLAttributes<HTMLDivElement> & StateProps & DispatchProps;
+
+const NotificationsDropdownWrapper = styled(Dropdown)``;
+
+const NotificationsDropdownHeader = styled.div`
+    align-items: center;
+    display: flex;
+    position: relative;
+`;
+
+const NewNotificationsBadge = styled.div`
+    background: #ff6534;
+    border-radius: 50%;
+    height: 7px;
+    position: absolute;
+    right: -4px;
+    top: 12px;
+    width: 7px;
+`;
+
+const NotificationsDropdownBody = styled(CardBase)`
+    width: 400px;
+`;
+const NotificationsList = styled.div`
+    height: 420px;
+    overflow: auto;
+`;
+
+const NotificationDropdownTitle = styled.h1`
+    border-bottom: 1px solid ${themeColors.borderColor};
+    color: #000;
+    font-size: 16px;
+    font-weight: 600;
+    line-height: 1.2;
+    margin: 0;
+    padding: 15px ${themeDimensions.horizontalPadding};
+`;
+
+class NotificationsDropdown extends React.Component<Props, {}> {
+    public render = () => {
+        const { notifications, hasUnreadNotifications, onMarkNotificationsAsRead, ...restProps } = this.props;
+
+        const notificationsList = notifications.map((item, index) => <NotificationItem key={index} item={item} />);
+
+        const header = (
+            <NotificationsDropdownHeader>
+                <BellIcon />
+                {hasUnreadNotifications ? <NewNotificationsBadge /> : null}
+            </NotificationsDropdownHeader>
+        );
+
+        const body = (
+            <NotificationsDropdownBody>
+                <NotificationDropdownTitle>Notifications</NotificationDropdownTitle>
+                <NotificationsList>{notificationsList}</NotificationsList>
```

Comment:
When there are no notifications, the dropdown's body opens anyway and shows a blank section.
I'm not 100% against it since (I believe) there are no designs for that scenario.

![no_notifications](https://user-images.githubusercontent.com/4421917/54053167-311f2c00-41c5-11e9-974c-ab62fbc47b26.png)

We can probably:
1. Keep it as it is,
2. Show a "No notifications" message inside the dropdown's body,
3. _Disable_ the dropdown by setting the `cursor: default;` + not showing the dropdown's body on click, 
4. Some other option that I can't think of right now.

I would pick option 2: I believe assigning a styled `<div>No notifications yet.</div>` to `notificationsList` if `notifications` is empty is enough (we'll have to ping @gabitoesmiapodo to check the style so it doesn't look _too bad_).

For option 3 I have an approach [here](https://github.com/unjapones/0x-launch-kit-frontend/blob/basic-notifications_disable-dropdown-when-no-notifications/src/components/notifications/notifications_dropdown.tsx), but I believe is too much.

Reply:
Also, if there are a few notifications (e.g. only 1), there is a white empty space on the dropdown's container.
If this is ok and expected, just ignore this comment.

![empty_space](https://user-images.githubusercontent.com/4421917/54053336-a854c000-41c5-11e9-9803-d7fd4690850e.png)

Reply:
+1 to showing a message.

Reply:
Couple of comments:
1. +1 to option 2: Show a "No notifications" message inside the dropdown's body.
1. Would it be too complex to limit dropdown size when there are just a few notifs?

Reply:
> Show a "No notifications" message inside the dropdown's body, 

+1

> Also, if there are a few notifications (e.g. only 1), there is a white empty space on the dropdown's container.

> Would it be too complex to limit dropdown size when there are just a few notifs?

Just change the list's container height from `height: 420px;` to `max-height: 420px;`

Reply:
Done. @gabitoesmiapodo please take a quick look at 3254386 to see if that way of showing the "No notifications" message is good enough. Here's how it looks:

![image](https://user-images.githubusercontent.com/417134/54130609-745ce300-43ef-11e9-8575-1f0752a1c30f.png)

Reply:
It's fine, I think.